src/main/modules/nunjucks/index.ts

```diff
@@ -27,9 +28,31 @@ export class Nunjucks {
       },
     );
 
+    env.addFilter('selectFilter', this.selectFilter);
+
     app.use((req, res, next) => {
       res.locals.pagePath = req.path;
       next();
     });
   }
+
+  private selectFilter(arr: SelectItem[], selectedId: string) {
+    // Set selected property on selected item
+    let itemSelected = false;
+    arr.forEach(si => {
+      if (si.value?.toString() === selectedId?.toString()) {
+        si.selected = true;
+        itemSelected = true;
+      } else {
+        si.selected = false;
+      }
+    });
+
+    if (!itemSelected) {
+      if (!arr.some(si => si.value === '')) {
+        arr.splice(0, 0, {value: '', text: '', selected: true});
```

Comment:
Might be worth commenting why we are doing this here; It isn't too clear to me. 

It looks like if we have any element that is not empty, splice the data, but I don't understand why without looking into it further! Might help other developer in the future to add a quick note in

Reply:
Comment now added.